lib/DxilDia/DxcPixLiveVariables.h

```diff
@@ -0,0 +1,83 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// DxcPixLiveVariables.h                                                     //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Implements a mapping from the instructions in the Module to the set of    //
+// live variables available in that instruction.                             //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+#include "dxc/Support/WinIncludes.h"
+
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "DxcPixDxilDebugInfo.h"
+
+namespace llvm
+{
+class DIVariable;
+class Instruction;
+class Module;
+class Value;
+}  // namespace llvm
+
+namespace dxil_debug_info
+{
+
+// VariableInfo is the bag with the information about a particular
+// DIVariable in the Module.
+struct VariableInfo
+{
+  using OffsetInBits = unsigned;
+
+  // Location is the dxil alloca register where this variable lives.
+  struct Location
+  {
+    llvm::Value *m_V = nullptr;
+    unsigned m_Offset = 0;
```

Comment:
OffsetInBits?

Reply:
Actually this is supposed to be the fragment index -- nice catch.